core-bundle/src/EventListener/SwitchUserListener.php

```diff
@@ -59,9 +60,25 @@ public function __invoke(SwitchUserEvent $event): void
             $targetUser = $targetUser->getUsername();
         }
 
-        $this->logger->info(
-            sprintf('User "%s" has switched to user "%s"', $sourceUser, $targetUser),
-            ['contao' => new ContaoContext(__METHOD__, ContaoContext::ACCESS, $sourceUser)]
-        );
+        $originalUser = null;
+
+        if ($token instanceof SwitchUserToken) {
+            $originalUser = $token->getOriginalToken()->getUser()->getUsername();
+        }
+
+        if ($originalUser === $targetUser) {
+            $this->logger->info(
+                sprintf('User "%s" has quit the impersonation of user "%s"', $originalUser, $sourceUser),
+                ['contao' => new ContaoContext(__METHOD__, ContaoContext::ACCESS, $originalUser)]
+            );
+        } else {
+            if (!empty($originalUser)) {
+                $sourceUser = $originalUser;
+            }
```

Comment:
```suggestion
            }

```

Reply:
I have currently no clue how allow edits of maintainers. I've added the new line after line 77 from the suggestion. But this runs into an "Cody styling error" in the checks.

Reply:
There's a checkbox on the lower right. After the closing bracket of the `if` and before `$this->logger->info` there should be an empty line.